### DISCUSSION TITLE
data fetch with context

### DISCUSSION THREAD

**Discussion by cjdxhjj** [graphql-java/graphql-java#3541] [Ideas]
the sub field may load with condition, but the data loader only load with key, for example the teacher has list of student, but i want to filter the student which live in newyork or anything else. i could wrap the id in an entity with the filter and combine the id and use the filter, but it need more work

**Comment by bbakerman** [comment]
Sorry I am not understanding your question.  I suspect you may be talking about aspects of

https://www.graphql-java.com/blog/deep-dive-data-fetcher-results

An example set of queries might help others understand your question

**Comment by cjdxhjj** [comment]
i have write the prefetch in my first version, but entity has many relation, because the datafetch could load with key batch, the performance is good

**Comment by cjdxhjj** [comment]
![image](https://github.com/graphql-java/graphql-java/assets/5616997/1175944a-aaf7-47ba-93e3-99baea80ac87)

**Comment by cjdxhjj** [comment]
i want to pass some other parameter to dataloader, for example systemId to filter the result, now i have to wrap the id and the extra parmater to an class
![image](https://github.com/graphql-java/graphql-java/assets/5616997/2d5c42dc-4d29-4944-a6ef-05478bc660db)
then i combine the id and get the first systemId(because the every systemId is same)

**Comment by cjdxhjj** [comment]
and then i fetch the result and group the entity by key and do some transfer to the result, for example, the dataloader require one key map to one entity, i have to wrap the result into an entity. that code is ugly

**Comment by cjdxhjj** [comment]
for my example, one user has many open id, the loader load with userid and filter with some parameter and group by userid and wrap to an entity has an list open id

**Comment by cjdxhjj** [comment]
i thought the dataloader can pass and retrive the context may give some help @bbakerman

**Comment by cjdxhjj** [comment]
![image](https://github.com/graphql-java/graphql-java/assets/5616997/34b809ef-bf7a-4a9f-8301-06400ce248c4)
![image](https://github.com/graphql-java/graphql-java/assets/5616997/a16ae98b-2b57-4150-8319-5fc1e0f52ec1)
the example is my solution, i wrap the query with an class and wrap the response with an class and group by id, when the data with id is missing i padding one, 
![image](https://github.com/graphql-java/graphql-java/assets/5616997/b1346777-6115-43cd-9e37-656fb711cc36)
i have write some method to padding one which onley fetch the known value from parent， for eample only fetch the id.

**Comment by cjdxhjj** [comment]
the list result wrapper can't avoid, and can wrapped by a common class for example pair in kotlin. when the dataloader provider the context, the idwrapper may avoid

**Comment by cjdxhjj** [comment]
![image](https://github.com/graphql-java/graphql-java/assets/5616997/e2e6dead-ae07-4fec-90d0-bc799ebd2e90)
the loader has only one method and can't get the context

**Comment by cjdxhjj** [comment]
![image](https://github.com/graphql-java/graphql-java/assets/5616997/d0c011c8-d985-4da2-a68e-752da8e2383b)

**Comment by cjdxhjj** [comment]
![image](https://github.com/graphql-java/graphql-java/assets/5616997/39c12bfb-ccf6-47e3-b4e3-8573d80b842b)

**Comment by cjdxhjj** [comment]
![image](https://github.com/graphql-java/graphql-java/assets/5616997/401be2cc-8c16-49ac-adf1-8c454ad6051d)

**Comment by cjdxhjj** [comment]
![image](https://github.com/graphql-java/graphql-java/assets/5616997/83d7aa66-4ac9-4eb6-b2de-e76081b52758)
